types/time.go

```diff
@@ -1020,7 +1020,9 @@ func parseDatetime(sc *stmtctx.StatementContext, str string, fsp int8, isFloat b
 
 			year, month, day, hour, minute, second =
 				dateTime.Year(), dateTime.Month(), dateTime.Day(), dateTime.Hour(), dateTime.Minute(), dateTime.Second()
-			if l >= 9 && l <= 14 {
+
+			// case: 0.XXX or like "20170118.999"
+			if seps[0] == "0" || l >= 9 && l <= 14 {
```

Comment:
Should we use `seps[0] == "0" || (l >= 9 && l <= 14)` here? It will look more clear.

Reply:
Exactly!